### PR TITLE
Remove HasValueSemantics from `is` ops.

### PR DESCRIPTION
**Pull request by silvasean**

### CODE REVIEW

**Comment by silvasean** [comment]
Part of the fix for https://github.com/llvm/torch-mlir/pull/993

File: python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/registry.py [hunk 1]
```diff
@@ -251,6 +251,11 @@ def has_value_semantics(self):
         # but the alias annotation is empty.
         if self.unique_key == "prim::unchecked_cast : (t) -> (t)":
             return False
+        # The `is` operator compares object identity, so it does not have
```
Comment (ashay) [inline python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/registry.py:254]: Nit: are you open to listing all overrides in one place?  I imagine we could need to add more in the future.

```python
overrides = [
        # It seems the FunctionSchema of "prim::unchecked_cast : (t) -> (t)" has
        # incorrect alias information. The result can alias with other tensors
        # but the alias annotation is empty.
        "prim::unchecked_cast : (t) -> (t)",

        # The `is` operator compares object identity, so it does not have
        # value semantics.
        "aten::__is__ : (t1, t2) -> (bool)",
        "aten::__isnot__ : (t1, t2) -> (bool)",
]

return not self.unique_key in overrides
```

**Review by ashay — APPROVED** [review]
Minor suggestion, but otherwise LGTM!

File: python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/registry.py (hunk 1 above)
Comment (silvasean) [reply python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/registry.py:254]: I think for the cases we have here, it reads a bit better to have them separate (if I put it in a list, I have a tendency to also want to hoist that out to the top level, which hurts reading locality). For sure if this list grows we will want something like you suggest.

**Pull request merged**